linode/terraform-provider-linode
# Support Nullable VPC Fields; Fix VPC Interface Update Error in `linode_instance` Resource

## 📝 Description

- Bump linodego version that contains the breaking change
- Handle VPC interface add/remove/swap issue in `linode_instance` resource.

## ✔️ How to Test

### Automated testing

`make PKG_NAME=linode/instance int-test`

### Manual testing

You can try removing, swapping (order), adding VPC interface to the Linode instance to verify if the fix works. 

```terraform
resource "linode_instance" "simple_instance" {
  label           = "simple_instance1441"
  image           = "linode/ubuntu22.04"
  region          = "us-mia"
  type            = "g6-nanode-1"
  
  interface {
    purpose   = "vpc"
    subnet_id = linode_vpc_subnet.foobar.id
    ipv4 {
      vpc = "10.0.4.250"
    }
  }
  
  interface {
    purpose = "public"
  }
}

resource "linode_vpc" "foobar" {
    label = "my-vpc"
    region = "us-mia"
    description = "test description"
}

resource "linode_vpc_subnet" "foobar" {
    vpc_id = linode_vpc.foobar.id
    label = "my-subnet"
    ipv4 = "10.0.4.0/24"
}
```
